### PR TITLE
feat(router): add app SSL config

### LINKED ISSUE #148: Enforce SSL on app-by-app basis
Currently, a cluster admin (if you subscribe to the Google operational model) can enable / enforce redirects from HTTP to HTTPS _router-wide_, but it seems that it would be tremendously useful to app admins (again, if you subscribe to the Google operational model) to opt into that on an app-by-app basis _if_ their cluster admin isn't already enforcing it router-wide.

### PR DESCRIPTION
**Pull request by bacongobbler**
Allow an application to enforce SSL by annotating `router.deis.io/nginx.ssl.enforce` on their service.

closes #148

TODO

 - [x] manual test
 - [x] README updates

### CODE REVIEW

File: nginx/config.go [hunk 1]
```diff
@@ -201,7 +201,7 @@ http {
 			proxy_set_header Upgrade $http_upgrade;
 			proxy_set_header Connection $connection_upgrade;
 
-			{{ if $enforceHTTPS }}if ($access_scheme != "https") {
+			{{ if or $enforceHTTPS $appConfig.SSL.Enforce }}if ($access_scheme != "https") {
 				return 301 https://$host$request_uri;
```
Comment (felixbuenemann) [inline nginx/config.go]: Shouldn't the redirect keep the query string? That can be done like shown below.

```
return 301 https://$host$request_uri$is_args$args;
```

File: nginx/config.go [hunk 2]
```diff
@@ -201,7 +201,7 @@ http {
 			proxy_set_header Upgrade $http_upgrade;
 			proxy_set_header Connection $connection_upgrade;
 
-			{{ if $enforceHTTPS }}if ($access_scheme != "https") {
+			{{ if or $enforceHTTPS $appConfig.SSL.Enforce }}if ($access_scheme != "https") {
```
Comment (felixbuenemann) [inline nginx/config.go]: Doesn't this mean that redirect can't be disabled on a per app basis, if it is enabled globally?

Comment (bacongobbler) [inline nginx/config.go]: That is correct and is intentional by design. The requirements are that if the router enforces something, it is not override-able by the user.

Precedence for this behaviour exists at https://github.com/deis/router/blob/master/nginx./config.go#L182

File: nginx/config.go (hunk 1 above)
Comment (bacongobbler) [inline nginx/config.go]: I'll leave that for @krancour to answer as I'm not as familiar with the router.

Comment (felixbuenemann) [inline nginx/config.go]: Yeah, it's not directly related to this PR, I just noticed it when reading the diff.

**Comment by bacongobbler** [comment]
Manual test:

```
><> curl go.k8s.local
Powered by Deis
Release v2 on go-web-4166859712-o5f90

><> k --namespace=go edit svc
service "go" edited

><> k --namespace=go get svc -o yaml | grep enforce
      router.deis.io/ssl.enforce: "True"

><> curl go.k8s.local
<html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx/1.11.2</center>
</body>
</html>
```

**Pull request merged**